vaultkeeper/tests/test_vaultkeeper.py

```diff
@@ -152,3 +167,81 @@ def test_renew_lease(self):
         })
         self.vaultkeeper.renew_lease(renew)
         assert renew.lease_duration == 300
+
+    @responses.activate
+    def test_run_natural_success(self, tmpdir):
+        self.vaultkeeper.vault_client.token = '00000000-0000-0000-0000-000000000001'
+        self.vaultkeeper.vault_secret = vault_token()
+        outfile = tmpdir.join(self.vaultkeeper.configs.credential_path)
+        self.vaultkeeper.configs.credential_path = outfile.strpath
+        responses.add_callback(responses.POST,
+                               self.fake_gatekeeper_url + '/token',
+                               callback=self.fake_gatekeeper.get_token,
+                               content_type='application/json')
```

Comment:
Maybe the fake gatekeeper needs helper methods to add these callbacks? `self.fake_gateway.add_handlers(responses, any, other, args)` would clean up some of the boilerplate.